### PR TITLE
MEN-5453

### PR DESCRIPTION
**Pull request by mzedel**

### CODE REVIEW

File: src/js/components/settings/integrations.js [hunk 1]
```diff
@@ -1,32 +1,119 @@
...
+      label={`${title} connection string`}
+      multiline
+      onChange={updateConnectionConfig}
+      value={value}
+    />
+  );
+};
+
+const providerConfigMap = {
+  amazon: ConnectionDetailsInput,
```
Comment (tranchitella) [inline src/js/components/settings/integrations.js:92]: This should be iot-core

**Review by tranchitella — CHANGES_REQUESTED** [review]

**Comment by tranchitella** [comment]
@mzedel still not working as expected, the UI is sending:
`{"provider":"amazon","credentials":{"type":"aws","access_key_id":"A","secret_access_key":"B","endpoint_url":"https://mender-test-hub.azure-devices.net","device_policy_ARN":"C"}}`

It should send:
`{"provider":"iot-hub","credentials":{"type":"aws","access_key_id":"A","secret_access_key":"B","endpoint_url":"https://mender-test-hub.azure-devices.net","device_policy_ARN":"C"}}`, instead.

**Comment by mzedel** [comment]
@tranchitella can you double check you tried it with the latest version? there should not be any trace of `amazon` in the src folder any more...

**Comment by tranchitella** [comment]
@mzedel I'm sorry, I messed up my local branch, you are right. I still have an issue, though: despite correct data being returned by the end-point, the UI is not populating the fields with the existing values:

image

Any idea why this is happening?

**Comment by mzedel** [comment]
@tranchitella yes you're right - the config destructuring and the casing for the config were not aligned with the endpoint - should now be aligned... I also added a simple URL validation for the endpoint, unfortunately it won't catch non-trusted domains (btw thanks a lot for the example data!)...

**Review by tranchitella — APPROVED** [review]

**Pull request merged**